packages/invoices/src/OutboundShipment/ListView/Toolbar.tsx

```diff
@@ -41,8 +44,28 @@ export const Toolbar: FC<{
     ref.current = deleteAction;
   }, [selectedRows]);
 
+  const key = 'otherPartyName' as keyof InvoiceRow;
+  const filterString = filter.filterBy?.[key]?.like;
+
   return (
-    <AppBarContentPortal sx={{ paddingBottom: '16px' }}>
+    <AppBarContentPortal
+      sx={{
+        paddingBottom: '16px',
+        flex: 1,
+        justifyContent: 'space-between',
+        display: 'flex',
+      }}
+    >
+      <BasicTextInput
+        value={filterString}
+        onChange={e =>
+          filter.onChangeStringFilterRule(
+            'otherPartyName',
+            'like',
+            e.target.value
+          )
+        }
+      />
```

Comment:
not sure what to do about the ui here

Reply:
looks ok, fairly subtle and out of the way. too much to ask to have a label on it?

Reply:
Not too much - i'll do the pagination fixes and the label in a new PR so I can trim this branches up!